### DISCUSSION TITLE
io.netty.handler.codec.http2.Http2Exception$StreamException: Stream closed before write could take place

### DISCUSSION THREAD

**Discussion by 1450793561** [apache/skywalking#12383] [Q&A]
The oap service restarts while running. The following error is found in the log. How can I solve the problem?

-----------------
<img width="1672" alt="企业微信截图_be1f57fa-c7fe-46eb-9120-8332459bdb9f" src="https://github.com/apache/skywalking/assets/31468018/a22c1b1f-fbac-42f0-bd7c-bc087869a78b">

skywalking-oap  | Jun 27, 2024 4:25:52 PM io.grpc.netty.NettyServerHandler onStreamError
skywalking-oap  | WARNING: Stream Error
skywalking-oap  | io.netty.handler.codec.http2.Http2Exception$StreamException: Stream closed before write could take place
skywalking-oap  |       at io.netty.handler.codec.http2.Http2Exception.streamError(Http2Exception.java:173)
skywalking-oap  |       at io.netty.handler.codec.http2.DefaultHttp2RemoteFlowController$FlowState.cancel(DefaultHttp2RemoteFlowController.java:481)
skywalking-oap  |       at io.netty.handler.codec.http2.DefaultHttp2RemoteFlowController$1.onStreamClosed(DefaultHttp2RemoteFlowController.java:105)
skywalking-oap  |       at io.netty.handler.codec.http2.DefaultHttp2Connection.notifyClosed(DefaultHttp2Connection.java:357)
skywalking-oap  |       at io.netty.handler.codec.http2.DefaultHttp2Connection$ActiveStreams.removeFromActiveStreams(DefaultHttp2Connection.java:1007)
skywalking-oap  |       at io.netty.handler.codec.http2.DefaultHttp2Connection$ActiveStreams.deactivate(DefaultHttp2Connection.java:963)
skywalking-oap  |       at io.netty.handler.codec.http2.DefaultHttp2Connection$DefaultStream.close(DefaultHttp2Connection.java:515)
skywalking-oap  |       at io.netty.handler.codec.http2.DefaultHttp2Connection.close(DefaultHttp2Connection.java:153)
skywalking-oap  |       at io.netty.handler.codec.http2.Http2ConnectionHandler$BaseDecoder.channelInactive(Http2ConnectionHandler.java:217)
skywalking-oap  |       at io.netty.handler.codec.http2.Http2ConnectionHandler.channelInactive(Http2ConnectionHandler.java:432)
skywalking-oap  |       at io.grpc.netty.NettyServerHandler.channelInactive(NettyServerHandler.java:637)
skywalking-oap  |       at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:305)
skywalking-oap  |       at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:281)
skywalking-oap  |       at io.netty.channel.AbstractChannelHandlerContext.fireChannelInactive(AbstractChannelHandlerContext.java:274)
skywalking-oap  |       at io.netty.channel.DefaultChannelPipeline$HeadContext.channelInactive(DefaultChannelPipeline.java:1405)
skywalking-oap  |       at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:301)
skywalking-oap  |       at io.netty.channel.AbstractChannelHandlerContext.invokeChannelInactive(AbstractChannelHandlerContext.java:281)
skywalking-oap  |       at io.netty.channel.DefaultChannelPipeline.fireChannelInactive(DefaultChannelPipeline.java:901)
skywalking-oap  |       at io.netty.channel.AbstractChannel$AbstractUnsafe$7.run(AbstractChannel.java:813)
skywalking-oap  |       at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:173)
skywalking-oap  |       at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:166)
skywalking-oap  |       at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470)
skywalking-oap  |       at io.netty.channel.epoll.EpollEventLoop.run(EpollEventLoop.java:413)
skywalking-oap  |       at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
skywalking-oap  |       at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
skywalking-oap  |       at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
skywalking-oap  |       at java.base/java.lang.Thread.run(Unknown Source)
skywalking-oap  |

**Comment by wu-sheng** [comment]
This is hard to tell. gRPC connection is broken somehow. I don't know how to fix it.
